common/include/pcl/common/impl/file_io.hpp

```diff
@@ -42,29 +42,30 @@
 namespace pcl
 {
 
-#ifndef _WIN32
-  void getAllPcdFilesInDirectory(const std::string& directory, std::vector<std::string>& file_names)
+void getAllPcdFilesInDirectory(const std::string& directory, std::vector<std::string>& file_names)
+{
+  boost::filesystem::path p(directory);
+  if(boost::filesystem::is_directory(p))
   {
-    DIR *dp;
-    struct dirent *dirp;
-    if((dp  = opendir(directory.c_str())) == nullptr) {
-      std::cerr << "Could not open directory.\n";
-      return;
-    }
-    while ((dirp = readdir(dp)) != nullptr) {
-      if (dirp->d_type == DT_REG)  // Only regular files
+    for(const auto& entry : boost::make_iterator_range(boost::filesystem::directory_iterator(p), {}))
+    {
+      if (boost::filesystem::is_regular_file(entry))
       {
-        std::string file_name = dirp->d_name;
-        if (file_name.substr(file_name.size()-4, 4)==".pcd")
-          file_names.emplace_back(dirp->d_name);
+        std::string file_name = entry.path().filename().string();
+        if (entry.path().extension() == "pcd")
```

Comment:
`boost::filesystem::path::extension()` outputs substring of the filename starting at the rightmost dot and ending at the path's end. This means you have to compare with `".pcd"` here.

Reply:
Thank for noticing this. I missed this 😅.